### PR TITLE
Testing homegrown feed emissions calculations

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR is to be used a testing ground for the homegrown feed emissions calculation routines defined in `emissions.py` (in the EEE module).

### Test plan

Run RuFaS with the below filters added. The homegrown feed emission metadata will be run as the only task. Note that I have shortened it significantly and turned the simulation of animals off to make it run quicker - to turn animal simulation on, change "simulate_animals" in c1f1_config.json. The emissions can still be observed for crops when not simulating animals, but animals will need to simulated to see the effect on purchased feed amounts and emissions.

These filters will give the emissions outputs and the numbers that `emissions.py` pulled from the Output Manager to calculate emissions, respectively:
```json
{
    "multiple": [
        {
            "name": "Actual Purchased Feeds",
            "filters": [
                ".*actual_purchased_feed_totals.*"
            ],
            "variables": [".*"]
        },
        {
            "name": "Actual Purchased Feed Emissions",
            "filters": [
                ".*actual_purchased_feed_emissions.*"
            ],
            "variables": [".*"]
        },
        {
            "name": "Homegrown Feed Emissions",
            "filters": [".*homegrown_.*_emissions.*"]
        }
    ]
}
```
and 
```json
{
  "multiple": [
    {
      "name": "DMI by feed",
      "filters": [
        "AnimalModuleReporter.report_daily_ration.ration_daily_feed_totals.*"
      ],
      "variables": [
        ".*"
      ],
      "vertical_aggregation": "sum",
      "slice_start": -365
    },
    {
      "name": "Feed emissions by feed",
      "filters": [
        "AnimalModuleReporter.report_daily_feed_emissions.*"
      ],
      "variables": [
        ".*"
      ],
      "vertical_aggregation": "sum",
      "slice_start": -365
    },
    {
      "name": "Yield",
      "filters": [
        "CropManagement._record_yield.harvest_yield.field='*'"
      ],
      "variables": [
          "crop",
          "wet_yield",
          "dry_yield",
          "nitrogen",
          "phosphorus",
          "harvest_index",
          "planting_date",
          "harvest_date",
          "residue_nitrogen",
          "yield_residue",
          "field_size"
        ]
    },
    {
      "name": "Soil N2O emissions",
      "filters": [
        "FieldDataReporter.send_daily_variables.nitrous_oxide_emissions.field=.*,layer=.*"
      ],
      "vertical_aggregation": "sum",
      "slice_start": -365
    },
    {
      "name": "Soil NH3 emissions",
      "filters": [
        "FieldDataReporter.send_daily_variables.ammonia_emissions.field=.*,layer=.*"
      ],
      "vertical_aggregation": "sum",
      "slice_start": -365
    },
    {
      "name": "profile_carbon_total-1",
      "filters": [
        "FieldDataReporter.send_daily_variables.total_soil_carbon_amount.field=.*"
      ],
      "slice_start": -1
    },
    {
      "name": "profile_carbon_total-365",
      "filters": [
        "FieldDataReporter.send_daily_variables.total_soil_carbon_amount.field=.*"
      ],
      "slice_start": -365,
      "slice_end": -364
    },
    {
      "name": "Fertilizer",
      "filters": [
        "Field._record_fertilizer_application\\.fertilizer_application\\.field='.*'"
      ],
      "variables": [".*"]
    }
  ]
}
```

### CODE REVIEW

**Comment by Isgotkowitz** [comment]
Update: I have verified some (but not all) of the estimated emissions and carbon stock change for the feeds in the homegrown feeds task. They were correct for the crops I tried, though I noticed that the alfalfa numbers I got manually were ~1% less than the estimated emissions. I assume this is due to rounding error.

**Comment by KFosterReed** [comment]
Meant to leave this comment in the PR, not the issue....

There was a duplicate feed ID number in the emissions.py file that had me scratching my head for a bit but I corrected it and then reviewed the corn silage, corn grain, and alfalfa silage actual purchased feed and emissions outputs... so far so good!

**Comment by KFosterReed** [comment]
Hi Ed! So I found a bug when testing a scenario where the farm did not grown enough feed and realized that there is a bug in the `calculated_actual_purchased_feeds` function because the amount of the homegrown_alternatives is reset with each new feed so instead of working with the updated value as set in this line: `homegrown_alternatives[key] -= amount_used`, it starts with the original value some how. I am not sure why though because when I read the code it seems to make sense to me.

**Comment by KFosterReed** [comment]
> Hi Ed! So I found a bug when testing a scenario where the farm did not grown enough feed and realized that there is a bug in the `calculated_actual_purchased_feeds` function because the amount of the homegrown_alternatives is reset with each new feed so instead of working with the updated value as set in this line: `homegrown_alternatives[key] -= amount_used`, it starts with the original value some how. I am not sure why though because when I read the code it seems to make sense to me.

To test again, I made a new homegrown feed metadata that uses the same field set up just with fewer acres. See the attached metadata and field input files
[homegrown_feed_test_metadata_LessHGFeed.json](https://github.com/user-attachments/files/15566906/homegrown_feed_test_metadata_LessHGFeed.json)
[c1f1_alfalfa_2021_cs_alf_field_SMALL.json](https://github.com/user-attachments/files/15566908/c1f1_alfalfa_2021_cs_alf_field_SMALL.json)
[c1f1_corngrain_2021_cg_alf_field_SMALL.json](https://github.com/user-attachments/files/15566909/c1f1_corngrain_2021_cg_alf_field_SMALL.json)
[c1f1_cornsilage_2021_cs_alf_field_SMALL.json](https://github.com/user-attachments/files/15566910/c1f1_cornsilage_2021_cs_alf_field_SMALL.json)

File: RUFAS/routines/EEE/emissions.py [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+        for feed_id, amount in purchased_feeds.items():
+            homegrown_alternatives = {
+                crop: crop_yield
+                for crop, crop_yield in homegrown_totals.items()
+                if feed_id in CROP_SPECIES_TO_PURCHASED_FEED_ID[crop]
+            }
+            for key, value in homegrown_alternatives.items():
+                amount_used = min(amount, value)
+                amount -= amount_used
+                homegrown_alternatives[key] -= amount_used
```
Comment (KFosterReed) [inline RUFAS/routines/EEE/emissions.py]: If this is changed to 

`homegrown_totals[key] -= amount_used` will that fix it?

Comment (KFosterReed) [reply RUFAS/routines/EEE/emissions.py]: I think it does

Comment (Isgotkowitz) [reply RUFAS/routines/EEE/emissions.py]: > I think it does

Nice fix Kristan. I added to it a little to remove some of the code that was made unnecessary by the fix

**Comment by Isgotkowitz** [comment]
Closed with the merging of PRs #1725 and #1750

**Pull request closed without merging**